File: framework/languages/java/e2e/SpotService/Shared/src/main/java/systems/zlink/e2e/spotservice/shared/Contracts.java
package systems.zlink.e2e.spotservice.shared;

import java.util.List;

public final class Contracts {
    public static final String ROUTE_CHANNEL = "spot.service.route";
    public static final String ROUTE_PACKET = "RouteReq";
    public static final String EGRESS_CHANNEL = "spot.service.egress";
    public static final String INGRESS_CHANNEL = "spot.service.ingress";
    public static final String SPOT_MESH = "spot.service.mesh";
    public static final String SPOT_NODE = "play";

    private Contracts() {
    }

    public record StateReq(String op) {
    }

    public record StateRes(
        String spotRid,
        String nodeRid,
        String value) {
    }

    public record StateMsg(String value) {
    }

    public record SpotStateOperation(
        String spotRid,
        String value,
        long timeoutMilliseconds) {
    }

    public record SpotValueOperation(
        String spotRid,
        String value) {
    }

    public record RouteOperation(
        String nodeRid,
        String value,
        long timeoutMilliseconds) {
    }

    public record ActorOperation(
        String actorId,
        String value,
        long timeoutMilliseconds) {
    }

    public record OperationAccepted(boolean accepted) {
    }

    public record MissingSpotReq(String value) {
    }

    public record MissingSpotMsg(String value) {
    }

    public record CreateSpotReq(String spotRid) {
    }

    public record CreateSpotRes(
        String spotRid,
        String nodeRid,
        String state) {
    }

    public record PlacementWeightReq(int weight) {
    }

    public record PlacementWeightRes(int weight) {
    }

    public record RelocationReq() {
    }

    public record RelocationRes(String outcome, String reason) {
    }

    public record SpotStateRouteReq(
        String spotRid,
        String op,
        int delta) {
    }

    public record SpotMissingHandlerReq(String spotRid) {
    }

    public record SpotMissingHandlerRes(
        String spotRid,
        boolean failed,
        EvidenceSnapshot evidence) {
    }

    public record SpotMissingCommandReq(
        String spotRid,
        String marker) {
    }

    public record SpotMissingCommandRes(
        String spotRid,
        String marker,
        boolean sent,
        EvidenceSnapshot evidence) {
    }

    public record StageProbeReq(
        String marker,
        int delta) {
    }

    public record SpotStageProbeRouteReq(
        String spotRid,
        String marker,
        int delta) {
    }

    public record StageTimerStartReq(
        String name,
        int periodMilliseconds) {
    }

    public record SpotStageTimerRouteReq(
        String spotRid,
        String name,
        int periodMilliseconds) {
    }

    public record StageTimerStartRes(
        String spotRid,
        String name,
        boolean started) {
    }

    public record MultiNodeCreateSpotReq(
        String spotRid,
        int delta) {
    }

    public record MultiNodeCreateSpotRes(
        String spotRid,
        String nodeRid,
        String state,
        int value) {
    }

    public record MultiNodeStateRouteReq(
        String spotRid,
        int delta) {
    }

    public record MultiNodeStateReq(int delta) {
    }

    public record MultiNodeStateMsg(String marker) {
    }

    public record MultiNodeStateRes(
        String spotRid,
        String nodeRid,
        int value) {
    }

    public record SpotOnlyMeshReq(
        String sourceSpotRid,
        String targetSpotRid,
        String marker) {
    }

    public record SpotOnlyMeshRes(
        String sourceSpotRid,
        String targetSpotRid,
        int targetValue,
        String marker) {
    }

    public record SpotOnlyJoinReq(
        String targetSpotRid,
        String actorId,
        String marker) {
    }

    public record SpotOnlyJoinRes(
        String targetSpotRid,
        String actorId,
        boolean accepted,
        String marker) {
    }

    public record SlowReq(String value) {
    }

    public record OutboundReq(String value) {
    }

    public record OutboundRes(
        String spotRid,
        String nodeRid,
        String channelReply) {
    }

    public record OutboundMsg(String value) {
    }

    public record MeshMsg(String value) {
    }

    public record TimerActivityReq(String value) {
    }

    public record TimerActivityRes(
        String spotRid,
        String value) {
    }

    public record RouteReq(String value) {
    }

    public record RouteRes(
        String value,
        String nodeRid,
        String routeRid) {
    }

    public record ActorProfile(
        String displayName,
        int level,
        List<String> tags) {
    }

    public record ActorAuthReq(
        String actorId,
        ActorProfile profile) {
    }

    public record ActorAuthRes(
        String actorId,
        String nodeRid,
        long generation,
        int boundCount,
        String displayName,
        int level,
        List<String> tags) {
    }

    public record MultiBindReq(
        String firstActorId,
        String secondActorId,
        ActorProfile profile) {
    }

    public record MultiBindRes(int boundCount) {
    }

    public record ActorJoinReq(
        String spotRid,
        ActorProfile profile,
        List<String> tags) {
    }

    public record JoinAdmittedUserSpotActorReq(
        String spotRid,
        ActorProfile profile,
        List<String> tags,
        boolean admit,
        String reason) {
    }

    public record ActorJoinRes(
        String actorId,
        String spotRid,
        String nodeRid,
        String displayName,
        int level,
        List<String> tags) {
    }

    public record JoinAdmittedUserSpotActorRes(
        String actorId,
        String spotRid,
        String nodeRid,
        boolean accepted,
        String errorKind) {
    }

    public record LeaveActorReq(String actorId) {
    }

    public record LeaveActorRes(
        String actorId,
        boolean accepted) {
    }

    public record ActorDestroyReq(String actorId) {
    }

    public record ActorDestroyRes(
        String actorId,
        boolean destroyed) {
    }

    public record ActorEchoReq(
        String value,
        int seq,
        ActorProfile profile) {
    }

    public record MissingActorReq(String value) {
    }

    public record ActorPingReq(String value) {
    }

    public record ActorPingRes(
        String actorId,
        String nodeRid,
        String spotRid,
        String value,
        int seen) {
    }

    public record ActorPushReq(String value) {
    }

    public record SnapshotReq(String actorId) {
    }

    public record SnapshotRes(
        String actorId,
        int seen) {
    }

    public record SlowSessionReq(
        String value,
        int delayMilliseconds) {
    }

    public record SlowSessionRes(String value) {
    }

    public record ActorEchoRes(
        String actorId,
        String spotRid,
        String nodeRid,
        String value,
        int requestSeq,
        int handlerSeq,
        String displayName,
        int level,
        List<String> tags) {
    }

    public record ActorPushNotify(
        String actorId,
        String spotRid,
        String value,
        int requestSeq,
        int handlerSeq) {
    }

    public record EvidenceEntry(
        String marker,
        String nodeRid,
        String spotRid,
        String value) {
    }

    public record EvidenceWaitReq(
        List<String> containsAll,
        int timeoutMilliseconds) {
    }

    public record EvidenceSnapshot(
        String nodeRid,
        List<EvidenceEntry> entries) {
    }

}

File: framework/languages/java/e2e-kotlin/SpotService/Shared/src/main/kotlin/systems/zlink/e2e/kotlin/spotservice/Contracts.kt
package systems.zlink.e2e.kotlin.spotservice

class Contracts private constructor() {
    companion object {
        const val ROUTE_CHANNEL: String = "spot.service.route"
        const val ROUTE_PACKET: String = "RoutePingReq"
        const val EGRESS_CHANNEL: String = "spot.service.egress"
        const val INGRESS_CHANNEL: String = "spot.service.ingress"
        const val SPOT_MESH: String = "spot.service.mesh"
        const val SPOT_NODE: String = "play"
        const val MULTI_SPOT_NODE_A: String = "multi-node-a"
        const val MULTI_SPOT_NODE_B: String = "multi-node-b"
        const val MULTI_ROUTE_CHANNEL_A: String = "multi-route-a"
        const val MULTI_ROUTE_CHANNEL_B: String = "multi-route-b"
        const val CHANNEL_HANDLER_GROUP: String = "spot-service-channel-handlers"
    }

    @JvmRecord
    data class StateReq(val op: String)

    @JvmRecord
    data class StateRes(
        val spotRid: String,
        val nodeRid: String,
        val value: String
    )

    @JvmRecord
    data class StateMsg(val value: String)

    @JvmRecord
    data class StageProbeReq(
        val marker: String,
        val op: String
    )

    @JvmRecord
    data class StageTimerStartMsg(
        val name: String,
        val periodMilliseconds: Int
    )

    @JvmRecord
    data class SpotStageProbeReq(
        val spotRid: String,
        val marker: String,
        val op: String
    )

    @JvmRecord
    data class SpotStageTimerReq(
        val spotRid: String,
        val name: String,
        val periodMilliseconds: Int
    )

    @JvmRecord
    data class SpotStageTimerRes(
        val spotRid: String,
        val name: String,
        val started: Boolean
    )

    @JvmRecord
    data class SlowReq(val value: String)

    @JvmRecord
    data class OutboundReq(val value: String)

    @JvmRecord
    data class OutboundRes(
        val spotRid: String,
        val nodeRid: String,
        val channelReply: String
    )

    @JvmRecord
    data class OutboundMsg(val value: String)

    @JvmRecord
    data class MeshMsg(val value: String)

    @JvmRecord
    data class TimerActivityReq(val value: String)

    @JvmRecord
    data class TimerActivityRes(
        val spotRid: String,
        val value: String
    )

    @JvmRecord
    data class TimerStatusRes(
        val spotRid: String,
        val value: String
    )

    @JvmRecord
    data class RoutePingReq(val value: String)

    @JvmRecord
    data class RoutePingRes(
        val value: String,
        val nodeRid: String,
        val routeRid: String
    )

    @JvmRecord
    data class SpotStateRouteReq(
        val spotRid: String,
        val op: String,
        val timeoutMilliseconds: Int = 5_000,
        val packetName: String = "StateReq"
    )

    @JvmRecord
    data class SpotStateCommandReq(
        val spotRid: String,
        val value: String,
        val packetName: String = "StateMsg"
    )

    @JvmRecord
    data class MissingSpotReq(val value: String)

    @JvmRecord
    data class MissingSpotMsg(val value: String)

    @JvmRecord
    data class CreateSpotReq(
        val spotRid: String
    )

    @JvmRecord
    data class CreateSpotRes(
        val spotRid: String,
        val nodeRid: String
    )

    @JvmRecord
    data class PlacementWeightReq(val weight: Int)

    @JvmRecord
    data class PlacementWeightRes(val weight: Int)

    @JvmRecord
    data class SpotSlowRouteReq(
        val spotRid: String,
        val value: String,
        val timeoutMilliseconds: Int
    )

    @JvmRecord
    data class SpotOutboundRouteReq(
        val spotRid: String,
        val value: String
    )

    @JvmRecord
    data class SpotOutboundCommandReq(
        val spotRid: String,
        val value: String,
        val packetName: String = "OutboundMsg"
    )

    @JvmRecord
    data class SpotToSpotCommandReq(
        val targetSpotRid: String,
        val value: String
    )

    @JvmRecord
    data class RoutePingHttpReq(
        val targetRid: String,
        val value: String
    )

    @JvmRecord
    data class AckRes(
        val accepted: Boolean
    )

    @JvmRecord
    data class MultiNodeCreateSpotReq(
        val spotRid: String,
        val delta: Int
    )

    @JvmRecord
    data class MultiNodeCreateSpotRes(
        val spotRid: String,
        val nodeRid: String,
        val state: String,
        val value: Int
    )

    @JvmRecord
    data class MultiNodeStateRouteReq(
        val spotRid: String,
        val delta: Int
    )

    @JvmRecord
    data class MultiNodeStateReq(
        val operation: String,
        val delta: Int
    )

    @JvmRecord
    data class SpotOnlyMeshReq(
        val sourceSpotRid: String,
        val targetSpotRid: String,
        val marker: String
    )

    @JvmRecord
    data class SpotOnlyMeshRes(
        val sourceSpotRid: String,
        val targetSpotRid: String,
        val targetValue: Int
    )

    @JvmRecord
    data class SpotOnlyJoinReq(
        val targetSpotRid: String,
        val actorId: String,
        val marker: String
    )

    @JvmRecord
    data class SpotOnlyJoinRes(
        val accepted: Boolean,
        val actorId: String
    )

    @JvmRecord
    data class MultiNodeStateRes(
        val spotRid: String,
        val nodeRid: String,
        val value: Int
    )

    @JvmRecord
    data class EvidenceWaitReq(
        val containsAll: List<String>,
        val timeoutMilliseconds: Int = 10_000
    )

    @JvmRecord
    data class ActorProfile(
        val displayName: String,
        val level: Int,
        val tags: List<String>
    )

    @JvmRecord
    data class ActorAuthReq(
        val actorId: String,
        val profile: ActorProfile
    )

    @JvmRecord
    data class ActorRemoteAuthReq(
        val actorId: String,
        val profile: ActorProfile,
        val nodeRid: String
    )

    @JvmRecord
    data class ActorAuthRes(
        val actorId: String,
        val nodeRid: String,
        val boundCount: Int,
        val displayName: String,
        val level: Int,
        val tags: List<String>
    )

    @JvmRecord
    data class EnsureActorReq(
        val actorId: String,
        val profile: ActorProfile
    )

    @JvmRecord
    data class EnsureActorRes(
        val actorId: String,
        val nodeRid: String,
        val generation: Long
    )

    @JvmRecord
    data class MultiBindReq(
        val firstActorId: String,
        val secondActorId: String,
        val profile: ActorProfile
    )

    @JvmRecord
    data class MultiBindRes(
        val boundCount: Int
    )

    @JvmRecord
    data class DestroyActorReq(
        val actorId: String
    )

    @JvmRecord
    data class DestroyActorRes(
        val actorId: String,
        val destroyed: Boolean
    )

    @JvmRecord
    data class LeaveActorReq(
        val actorId: String
    )

    @JvmRecord
    data class LeaveActorRes(
        val actorId: String,
        val accepted: Boolean
    )

    @JvmRecord
    data class ActorJoinReq(
        val spotRid: String,
        val profile: ActorProfile,
        val tags: List<String>
    )

    @JvmRecord
    data class ActorJoinRes(
        val actorId: String,
        val spotRid: String,
        val nodeRid: String,
        val displayName: String,
        val level: Int,
        val tags: List<String>
    )

    @JvmRecord
    data class ActorEchoReq(
        val value: String,
        val seq: Int,
        val profile: ActorProfile
    )

    data class MissingActorReq(
        val value: String,
        val seq: Int,
        val profile: ActorProfile,
    )

    @JvmRecord
    data class SlowSessionReq(
        val value: String,
        val delayMilliseconds: Int
    )

    @JvmRecord
    data class SlowSessionRes(
        val value: String
    )

    @JvmRecord
    data class ActorEchoRes(
        val actorId: String,
        val spotRid: String,
        val nodeRid: String,
        val value: String,
        val requestSeq: Int,
        val handlerSeq: Int,
        val displayName: String,
        val level: Int,
        val tags: List<String>
    )

    @JvmRecord
    data class ActorPushNotify(
        val actorId: String,
        val spotRid: String,
        val value: String,
        val requestSeq: Int,
        val handlerSeq: Int
    )

    @JvmRecord
    data class EvidenceEntry(
        val marker: String,
        val nodeRid: String,
        val spotRid: String,
        val value: String?
    )

    @JvmRecord
    data class EvidenceSnapshot(
        val nodeRid: String,
        val entries: List<EvidenceEntry>
    )
}
